.github/workflows/memory-check.yml

```
@@ -0,0 +1,157 @@
+          pip install maturin memory_profiler
+      
+      - name: Build Rust extension
+        run: |
+          source .venv/bin/activate
+          maturin develop --release
+      
+      - name: Run Rust memory checks with Valgrind
+        run: |
+          # Run Rust tests under valgrind
+          cargo test --release 2>&1 | tee rust-test.log
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)